pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1050,27 +1047,55 @@ func (n *NetworkPolicyController) syncAddressGroup(key string) error {
 			pods = append(pods, nsPods...)
 		}
 	}
-	addresses := sets.String{}
+	podSet := networking.GroupMemberPodSet{}
 	for _, pod := range pods {
 		if pod.Status.PodIP == "" {
 			// No need to insert Pod IPAdddress when it is unset.
 			continue
 		}
-		addresses.Insert(pod.Status.PodIP)
+		podSet.Insert(podToMemberPod(pod, true, false))
 	}
 	updatedAddressGroup := &antreatypes.AddressGroup{
-		Name:      addressGroup.Name,
-		UID:       addressGroup.UID,
-		Selector:  addressGroup.Selector,
-		Addresses: addresses,
-		SpanMeta:  spanMeta,
+		Name:     addressGroup.Name,
+		UID:      addressGroup.UID,
+		Selector: addressGroup.Selector,
+		Pods:     podSet,
+		SpanMeta: spanMeta,
 	}
-	klog.V(2).Infof("Updated AddressGroup %s with addresses %v and Node names %v", key, addresses, addrGroupNodeNames)
+	klog.V(2).Infof("Updated AddressGroup %s with addresses %v and Node names %v", key, podSet, addrGroupNodeNames)
 	// Update the store of AddressGroup.
 	n.addressGroupStore.Update(updatedAddressGroup)
 	return nil
 }
 
+func podToMemberPod(pod *v1.Pod, includeIP, includePodRef bool) *networking.GroupMemberPod {
+	memberPod := &networking.GroupMemberPod{}
+	for _, container := range pod.Spec.Containers {
+		for _, port := range container.Ports {
+			// Only include container ports with name set.
+			if len(port.Name) > 0 {
```

Comment:
curious why you chose len instead of port.Name != "" ?

Reply:
Just picked one casually, changed to `port.Name != ""` if it's cheaper.